viam-orb-slam3.go

```diff
@@ -747,108 +707,82 @@ func (slamSvc *builtIn) StartSLAMProcess(ctx context.Context) error {
 				}
-				slamSvc.port = "localhost:" + strings.TrimRight(linePieces[1], "\n")
+				orbSvc.port = "localhost:" + strings.TrimRight(linePieces[1], "\n")
 				break
 			}
 		}
 	}
 
-	if slamSvc.bufferSLAMProcessLogs {
-		slamSvc.slamProcessLogReader = logReader
-		slamSvc.slamProcessLogWriter = logWriter
-		slamSvc.slamProcessBufferedLogReader = bufferedLogReader
+	if orbSvc.bufferSLAMProcessLogs {
+		orbSvc.slamProcessLogReader = logReader
+		orbSvc.slamProcessLogWriter = logWriter
+		orbSvc.slamProcessBufferedLogReader = bufferedLogReader
 	}
 
 	return nil
 }
 
 // stopSLAMProcess uses the process manager to stop the created slam process from running.
-func (slamSvc *builtIn) StopSLAMProcess() error {
-	if err := slamSvc.slamProcess.Stop(); err != nil {
+func (orbSvc *orbslamService) StopSLAMProcess() error {
+	if err := orbSvc.slamProcess.Stop(); err != nil {
 		return errors.Wrap(err, "problem stopping slam process")
 	}
 	return nil
 }
 
-func (slamSvc *builtIn) getPNGImage(ctx context.Context, cam camera.Camera) ([]byte, func(), error) {
-	// We will hint that we want a PNG.
-	// The Camera service server implementation in RDK respects this; others may not.
-	readImgCtx := gostream.WithMIMETypeHint(ctx, rdkutils.WithLazyMIMEType(rdkutils.MimeTypePNG))
-	img, release, err := camera.ReadImage(readImgCtx, cam)
-	if err != nil {
-		return nil, nil, err
-	}
-	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
-		if lazyImg.MIMEType() != rdkutils.MimeTypePNG {
-			return nil, nil, errors.Errorf("expected mime type %v, got %T", rdkutils.MimeTypePNG, img)
-		}
-		return lazyImg.RawData(), release, nil
-	}
-
-	if ycbcrImg, ok := img.(*image.YCbCr); ok {
-		pngImage, err := rimage.EncodeImage(ctx, ycbcrImg, rdkutils.MimeTypePNG)
-		if err != nil {
-			return nil, nil, err
-		}
-		return pngImage, release, nil
-	}
-
-	return nil, nil, errors.Errorf("expected lazily encoded image or ycbcrImg, got %T", img)
-}
-
-// getAndSaveDataSparse implements the data extraction for sparse algos and saving to the directory path (data subfolder) specified in
+// getAndSaveData implements the data extraction for saving to the directory path (data subfolder) specified in
 // the config. It returns the full filepath for each file saved along with any error associated with the data creation or saving.
-func (slamSvc *builtIn) getAndSaveDataSparse(
+func (orbSvc *orbslamService) getAndSaveData(
 	ctx context.Context,
 	cams []camera.Camera,
 ) ([]string, error) {
-	ctx, span := trace.StartSpan(ctx, "slam::builtIn::getAndSaveDataSparse")
+	ctx, span := trace.StartSpan(ctx, "slam::orbslamService::getAndSaveDataSparse")
 	defer span.End()
 
-	switch slamSvc.slamMode {
-	case slam.Mono:
+	switch orbSvc.subAlgo {
+	case Mono:
 		if len(cams) != 1 {
 			return nil, errors.Errorf("expected 1 camera for mono slam, found %v", len(cams))
 		}
 
-		image, release, err := slamSvc.getPNGImage(ctx, cams[0])
+		image, release, err := slamSensorUtils.GetPNGImage(ctx, cams[0])
```

Comment:
This is a great change - eventually, we'll want to use the RGB's `GetData` function.